compiler/src/iree/compiler/Utils/ShapeUtils.h

```diff
@@ -0,0 +1,21 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_UTILS_SHAPEUTILS_H_
+#define IREE_COMPILER_UTILS_SHAPEUTILS_H_
+
+#include "mlir/IR/BuiltinTypeInterfaces.h"
+#include "mlir/IR/ValueRange.h"
+
+namespace mlir::iree_compiler {
+
+/// Helper to compare shapes of two shaped types by SSA equivalence.
+bool compareShapesEqual(ShapedType lhsType, ValueRange lhsDynamicDims,
+                        ShapedType rhsType, ValueRange rhsDynamicDims);
```

Comment:
Every time I see this kind of method, I'd think if it is a soft check or hard check. I.e., I wonder if it returns true or not when one dimension is dynamic and the other is static. Maybe we can add such information to the name (but I don't have a good name yet), or we can add the information to the comment.

For this one, it returns false if one is dynamic and the other is static.